core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -145,30 +156,30 @@ public void commitRecords(Snapshot snapshot) {
     }
   }
 
-  public TransactionState abort(String id) throws UnknownTransactionStatusException {
+  public TransactionState abortState(String id) throws UnknownTransactionStatusException {
     try {
-      abortState(id);
+      Coordinator.State state = new Coordinator.State(id, TransactionState.ABORTED);
+      coordinator.putState(state);
       return TransactionState.ABORTED;
-    } catch (CoordinatorException e) {
+    } catch (CoordinatorConflictException e) {
```

Comment:
Similar to `commitState()`, I think we should check the state only when catching `CoordinatorConflictException`.